src/operator/pooling-inl.h

```diff
@@ -57,13 +57,13 @@ struct PoolingParam : public dmlc::Parameter<PoolingParam> {
               "kValid is default setting of Mxnet and rounds down the output pooling size."
               "kFull is compatible with Caffe and rounds up the output pooling size.");
 
-    int stride_shape[] = {1, 1};
-    DMLC_DECLARE_FIELD(stride).set_default(TShape(stride_shape, stride_shape + 2))
+    int stride_shape[] = {1, 1, 1};
+    DMLC_DECLARE_FIELD(stride).set_default(TShape(stride_shape, stride_shape + kernel.ndim()))
```

Comment:
Can I set shapes in PoolingProp::InferShape? It is a const function.

I can set them in PoolingProp::Init as convolution-inl.h does. 
https://github.com/dmlc/mxnet/blob/898c4e71f1d21aae2cae5a86eb9ed15c1c01ada1/src/operator/convolution-inl.h#L354